### ISSUE TITLE
Move CCLib to a separate project

### ISSUE THREAD

**Issue by ElvishJerricco** [Team-CC-Corp/JVML-JIT#73]
Putting CCLib under [GPL with classpath exception](http://openjdk.java.net/legal/gplv2+ce.html) would enable us to use source code from OpenJDK, without forcing JVML-JIT, or users of JVML-JIT to adopt the GPL.

**Comment by ElvishJerricco** [comment]
@Yevano, what do you think about this idea?

**Comment by JBYoshi** [comment]
This seems like a good idea. It could lead to simply using most of the same code, with only the native methods needing to be reimplemented (plus a few other changes, such as adding CC interaction code).

**Comment by ElvishJerricco** [comment]
Additionally, it should be noted, we can't expect to use OpenJDK's actual rt.jar in the distant future, because it is far too many megabytes in size for most CC space limits. We will need to be using a custom, leaner rt.jar.

**Comment by JBYoshi** [comment]
Right. However, we could trim out some classes that can't be implemented in CC (e.g. AWT).

**Comment by Turakar** [comment]
Also have a look at #25, where I suggested the seperation and license change.
